<a href="https://colab.research.google.com/github/juliosdutra/Controle_de_Processos_UFES/blob/main/2.Model-linearization%20/Lineariza%C3%A7%C3%A3o_usando_Sympy.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<left><img src="https://github.com/juliosdutra/Controle_de_Processos_UFES/blob/main/imagens/cabe%C3%A7alho-geral.jpg?raw=true" width="700" height="175">
</left>

**Lista 2 / Questão # 3**

Um poluente pode ser descartado somente se o efluente apresentar no máximo
5 ppm. Para tanto, é usado um reator agitado, operado com volume, densidade e temperatura constantes, cujo modelo dinâmico é:

$$ \frac{dCa}{dt} = D(t)\cdot[Ca_{f}(t)-Ca] - Ra $$

O poluente é alimentado com taxa de diluição D e concentração Caf e tem a seguinte taxa de reação: $$Ra = 15Ca/(1+0.1Ca)~\mathrm{[ppm\cdot h^{-1}]}$$

A condição estacionária é dada por $D_s=4.95~\mathrm{h^{-1}}$, $Ca_{f,s}=10 ~\mathrm{ppm}$ e $Ca_s=3.0~\mathrm{ppm}$.

Neste contexto, determine o modelo linear em variáveis-desvio.

In [ ]:
import sympy as sp
D, Caf, Ca = sp.symbols('D Caf Ca')
Ra = 15*Ca/(1+0.1*Ca)
f_nao_linear = D*(Caf-Ca) - Ra
f_nao_linear

-15*Ca/(0.1*Ca + 1) + D*(-Ca + Caf)

In [ ]:
# Dicionário com a condição estacionária
nominal = {'Ca': 3, 'D': 4.95, 'Caf': 10}

In [ ]:
# Derivada de f em relação a Ca
f_nao_linear.diff(Ca).simplify()

-D - 15/(0.1*Ca + 1)**2

In [ ]:
# Derivada de f em relação a D
f_nao_linear.diff(D).simplify()

-Ca + Caf

In [ ]:
# Derivada de f em relação a Caf
f_nao_linear.diff(Caf).simplify()

D

In [ ]:
# Calculando o valor das derivadas no ponto nominal
f_Cas =  f_nao_linear.diff(Ca).subs(nominal)
f_Cafs =  f_nao_linear.diff(Caf).subs(nominal)
f_Ds =  f_nao_linear.diff(D).subs(nominal)

In [ ]:
# Expressão de f linearizada
f_linear = f_Cas*Ca + f_Cafs*Caf + f_Ds*D
f_linear

-13.8257396449704*Ca + 4.95*Caf + 7*D

A partir deste ponto, pode-se escrever que

$$ \frac{dCa}{dt}  = -13.826 Ca + 4.95 Caf + 7 D $$

Na forma padrão:

$ \frac{dCa}{dt} +13.826 Ca =  4.95 Caf + 7 D $
<br>
<br>
$ \frac{1}{13.826} \frac{dCa}{dt} + Ca =  \frac{4.95}{13.826} Caf + \frac{7}{13.826} D $
<br>
<br>
Finalmente, em termo de variáveis-desvio, o modelo linearizado é:
<br>

$$ 0.072 \frac{dCa}{dt} + Ca =  0.358 Caf + 0.506 D $$
$$Ca(0) = 0$$

